src/merkle_tree.rs

```diff
@@ -0,0 +1,206 @@
+use hmac_sha256::Hash;
+
+pub struct MerkleTree {
+    pub root: Option<[u8; 32]>,
+    pub leaves: Vec<[u8; 32]>,
+}
+
+impl MerkleTree {
+    /// Create a new empty MerkleTree instance.
+    pub fn new() -> Self {
+        Self {
+            root: None,
+            leaves: Vec::new(),
+        }
+    }
+
+    /// Create a new MerkleTree from the provided items.
+    /// Each item should be representable as bytes.
+    /// It returns a `MerkleTree` instance with the leaf hashes and the Merkle root.
+    pub fn build<T: AsRef<[u8]>>(items: &[T]) -> Self {
+        if items.is_empty() {
+            return Self::new();
+        }
+
+        let leaves: Vec<[u8; 32]> = items.iter().map(|item| Self::hash(item.as_ref())).collect();
+
+        Self {
+            root: Some(Self::merkle_root(leaves.clone())),
+            leaves,
+        }
+    }
```

Comment:
I'd like to have someone else's output on this, but in my head rather than having an empty tree case, I'd have all trees having something inside, and `build` return an `Option<MerkleTree>`, where passing an empty slice produces `None`.

Reply:
We talked a bit about this [here](https://github.com/lazcanoluca/merkle-tree/pull/1#discussion_r1694438280), where we concluded it would be okay to build an empty Merkle Tree from an empty slice. I agree using an `Option` makes total sense. I think I'm going with the `Option`.